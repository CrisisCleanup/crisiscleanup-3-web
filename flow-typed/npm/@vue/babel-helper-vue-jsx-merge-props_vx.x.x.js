// flow-typed signature: d7a23c53f4091ef2bf1df8afa5eb43c4
// flow-typed version: <<STUB>>/@vue/babel-helper-vue-jsx-merge-props_v^1.0.0/flow_v0.131.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/babel-helper-vue-jsx-merge-props'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/babel-helper-vue-jsx-merge-props' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/babel-helper-vue-jsx-merge-props/dist/helper' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/babel-helper-vue-jsx-merge-props/dist/helper.js' {
  declare module.exports: $Exports<'@vue/babel-helper-vue-jsx-merge-props/dist/helper'>;
}
