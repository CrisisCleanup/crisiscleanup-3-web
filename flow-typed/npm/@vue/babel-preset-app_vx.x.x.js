// flow-typed signature: 6ba91a92845d9a556a5bcdcc4f033fed
// flow-typed version: <<STUB>>/@vue/babel-preset-app_v^4.3.1/flow_v0.129.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/babel-preset-app'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/babel-preset-app' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/babel-preset-app/polyfillsPlugin' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/babel-preset-app/index' {
  declare module.exports: $Exports<'@vue/babel-preset-app'>;
}
declare module '@vue/babel-preset-app/index.js' {
  declare module.exports: $Exports<'@vue/babel-preset-app'>;
}
declare module '@vue/babel-preset-app/polyfillsPlugin.js' {
  declare module.exports: $Exports<'@vue/babel-preset-app/polyfillsPlugin'>;
}
