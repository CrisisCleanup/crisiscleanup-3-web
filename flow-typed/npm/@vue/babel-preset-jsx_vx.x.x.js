// flow-typed signature: eeecaeff29e4a3a6c63c4fe7e16a0eb3
// flow-typed version: <<STUB>>/@vue/babel-preset-jsx_v^1.1.2/flow_v0.131.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/babel-preset-jsx'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/babel-preset-jsx' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/babel-preset-jsx/dist/plugin.cjs' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/babel-preset-jsx/dist/plugin.cjs.js' {
  declare module.exports: $Exports<'@vue/babel-preset-jsx/dist/plugin.cjs'>;
}
