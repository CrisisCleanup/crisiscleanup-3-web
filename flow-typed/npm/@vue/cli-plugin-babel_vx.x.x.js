// flow-typed signature: 4cf1a98e80b05c0a167e013d64bc9645
// flow-typed version: <<STUB>>/@vue/cli-plugin-babel_v^4.4.6/flow_v0.131.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/cli-plugin-babel'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/cli-plugin-babel' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/customConfig.input' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/customConfig.output' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/default.input' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/default.output' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/doubleQuote.input' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/doubleQuote.output' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/require.input' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/require.output' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/templateLiteral.input' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/templateLiteral.output' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/usePluginPreset' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/generator' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/migrator' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/preset' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/customConfig.input.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/customConfig.input'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/customConfig.output.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/customConfig.output'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/default.input.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/default.input'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/default.output.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/default.output'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/doubleQuote.input.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/doubleQuote.input'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/doubleQuote.output.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/doubleQuote.output'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/require.input.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/require.input'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/require.output.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/require.output'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/templateLiteral.input.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/templateLiteral.input'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/templateLiteral.output.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/templateLiteral.output'>;
}
declare module '@vue/cli-plugin-babel/codemods/usePluginPreset.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/usePluginPreset'>;
}
declare module '@vue/cli-plugin-babel/generator.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/generator'>;
}
declare module '@vue/cli-plugin-babel/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel'>;
}
declare module '@vue/cli-plugin-babel/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel'>;
}
declare module '@vue/cli-plugin-babel/migrator/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/migrator'>;
}
declare module '@vue/cli-plugin-babel/migrator/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/migrator'>;
}
declare module '@vue/cli-plugin-babel/preset.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/preset'>;
}
