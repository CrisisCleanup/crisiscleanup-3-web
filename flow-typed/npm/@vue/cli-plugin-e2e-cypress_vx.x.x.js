// flow-typed signature: 122252e4d027555936ded476aff636b4
// flow-typed version: <<STUB>>/@vue/cli-plugin-e2e-cypress_v^4.4.6/flow_v0.129.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/cli-plugin-e2e-cypress'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/cli-plugin-e2e-cypress' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/cli-plugin-e2e-cypress/generator' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/_eslintrc' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/specs/test' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/commands' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/ui' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/cli-plugin-e2e-cypress/generator/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/_eslintrc.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/_eslintrc'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/specs/test.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/specs/test'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/commands.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/commands'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support'>;
}
declare module '@vue/cli-plugin-e2e-cypress/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress'>;
}
declare module '@vue/cli-plugin-e2e-cypress/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress'>;
}
declare module '@vue/cli-plugin-e2e-cypress/ui.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/ui'>;
}
