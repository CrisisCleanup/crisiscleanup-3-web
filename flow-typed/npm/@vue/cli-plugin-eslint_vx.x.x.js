// flow-typed signature: 190fd4925231c8aa9cbfe00ac40be4f1
// flow-typed version: <<STUB>>/@vue/cli-plugin-eslint_v^4.4.6/flow_v0.129.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/cli-plugin-eslint'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/cli-plugin-eslint' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/cli-plugin-eslint/eslintDeps' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-eslint/eslintOptions' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-eslint/generator' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-eslint/lint' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-eslint/migrator' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-eslint/prompts' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-eslint/ui/configDescriptor' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-eslint/ui' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-eslint/ui/taskDescriptor' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/cli-plugin-eslint/eslintDeps.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/eslintDeps'>;
}
declare module '@vue/cli-plugin-eslint/eslintOptions.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/eslintOptions'>;
}
declare module '@vue/cli-plugin-eslint/generator/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/generator'>;
}
declare module '@vue/cli-plugin-eslint/generator/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/generator'>;
}
declare module '@vue/cli-plugin-eslint/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint'>;
}
declare module '@vue/cli-plugin-eslint/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint'>;
}
declare module '@vue/cli-plugin-eslint/lint.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/lint'>;
}
declare module '@vue/cli-plugin-eslint/migrator/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/migrator'>;
}
declare module '@vue/cli-plugin-eslint/migrator/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/migrator'>;
}
declare module '@vue/cli-plugin-eslint/prompts.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/prompts'>;
}
declare module '@vue/cli-plugin-eslint/ui/configDescriptor.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/ui/configDescriptor'>;
}
declare module '@vue/cli-plugin-eslint/ui/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/ui'>;
}
declare module '@vue/cli-plugin-eslint/ui/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/ui'>;
}
declare module '@vue/cli-plugin-eslint/ui/taskDescriptor.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/ui/taskDescriptor'>;
}
