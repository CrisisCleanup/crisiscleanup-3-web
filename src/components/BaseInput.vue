<template>
  <div class="flex justify-start relative" :class="classes">
    <label v-if="topLabel" class="text-xs px-1 text-crisiscleanup-dark-300">{{
      topLabel
    }}</label>
    <component
      :is="textArea ? 'textarea' : 'input'"
      ref="input"
      :class="[inputClasses, defaultInputClasses, selector]"
      :style="[cssVars, inputStyle]"
      :type="passwordView || type || 'search'"
      :value.prop="modelValue"
      :disabled="disabled || (breakGlass && !glassBroken)"
      :placeholder="placeholder"
      :required="required"
      :hidden="hidden"
      :pattern="pattern"
      :autocomplete="autocomplete"
      :rows="rows"
      @input="update"
      @input.stop=""
      @change="change"
      @blur="$emit('blur', $event)"
      @focus="$emit('focus', $event)"
      @keyup.enter="$emit('enter')"
      @invalid="isInvalid = true"
    />
    <div
      v-if="breakGlass && !glassBroken && !hidden"
      class="icon-container flex items-center justify-center"
      :class="iconClasses"
      @click="glassBroken = true"
    >
      <ccu-icon
        :alt="$t('actions.edit')"
        data-testid="testBreakGlassButton"
        type="edit"
        size="small"
        class="js-break-glass"
      />
    </div>
    <div
      v-if="faIcon"
      class="icon-container flex items-center justify-center"
      :class="iconClasses"
    >
      <ccu-icon
        :fa="true"
        size="medium"
        :type="faIcon"
        :alt="tooltip"
        :action="
          () => {
            $emit('iconClicked');
          }
        "
      />
    </div>
    <div
      v-else-if="icon || tooltip"
      v-tooltip="{
        content: tooltip,
        triggers: ['hover'],
        popperClass: 'interactive-tooltip w-72',
        html: true,
      }"
      class="icon-container flex items-center justify-center"
      :class="iconClasses"
    >
      <ccu-icon
        :alt="$t('actions.help_alt')"
        :type="tooltip ? 'info' : icon"
        :size="iconSize"
      />
    </div>
    <font-awesome-icon
      v-if="type === 'password'"
      :alt="$t('actions.help_alt')"
      size="lg"
      class="cursor-pointer absolute right-0 mr-4"
      :icon="passwordView === 'text' ? 'eye-slash' : 'eye'"
      @click="
        () => {
          if (passwordView === 'text') {
            passwordView = 'password';
          } else {
            passwordView = 'text';
          }
        }
      "
    />
    <slot></slot>
  </div>
</template>

<script lang="ts">
import { ref, computed, defineComponent } from 'vue';

export default defineComponent({
  name: 'BaseInput',
  props: {
    size: {
      type: String,
      default: '',
    },
    icon: {
      type: String,
      default: '',
    },
    faIcon: {
      type: String,
      default: '',
    },
    iconSize: {
      type: String,
      default: 'small',
    },
    modelValue: {
      type: null,
      default: null,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    hidden: {
      type: Boolean,
      default: false,
    },
    placeholder: {
      type: String,
      default: '',
    },
    pattern: {
      type: String,
      default: '.*',
    },
    required: {
      type: Boolean,
      default: false,
    },
    tooltip: {
      type: String,
      default: '',
    },
    topLabel: {
      type: String,
      default: '',
    },
    type: {
      type: String,
      default: '',
    },
    inputStyle: {
      type: String,
      default: '',
    },
    inputClasses: {
      type: String,
      default: '',
    },
    breakGlass: {
      type: Boolean,
      default: false,
    },
    readonly: {
      type: Boolean,
      default: false,
    },
    selector: {
      type: String,
      default: '',
    },
    autocomplete: {
      type: String,
      default: 'off',
    },
    textArea: {
      type: Boolean,
      default: false,
    },
    rows: {
      type: Number,
      default: 0,
    },
    validator: {
      type: [Function, Boolean],
      default: false,
    },
    width: {
      type: String,
      default: '300',
    },
    height: {
      type: String,
      default: '40',
    },
  },
  emits: ['update:modelValue'],

  setup(props, context) {
    const id = ref(null);
    const isInvalid = ref(false);
    const input = ref(null);
    const passwordView = ref(props.type === 'password' ? 'password' : '');
    const iconClasses = ref({
      large: props.size === 'large',
      small: props.size === 'small',
      base: props.size !== 'large' && props.size !== 'small',
      'has-tooltip': Boolean(props.tooltip),
    });
    const glassBroken = ref(false);

    const cssVars = computed(() => {
      return {
        '--height': `${props.height}px`,
        '--width': `${props.width}px`,
      };
    });

    const classes = computed(() => ({
      'border-crisiscleanup-red-100': props.topLabel && isInvalid.value,
      'flex-col items-start border': Boolean(props.topLabel),
      'items-center': !props.topLabel,
    }));

    const defaultInputClasses = computed(() => ({
      'flex-grow': true,
      'p-1': true,
      'text-base': !props.inputClasses,
      xlarge: props.size === 'xlarge',
      large: props.size === 'large',
      medium: props.size === 'medium',
      small: props.size === 'small',
      base: props.size === 0,
      'has-icon': Boolean(props.icon),
      'has-tooltip': Boolean(props.tooltip),
      invalid: Boolean(isInvalid.value),
      'border-none': Boolean(props.topLabel),
    }));

    function update(e) {
      const { value } = e.target;
      if (props.validator) {
        const { newValue, valid } = props.validator(value);
        isInvalid.value = valid;
        if (newValue) {
          input.value = newValue;
          return context.emit('update:modelValue', newValue);
        }
      }

      context.emit('update:modelValue', value);
      isInvalid.value = !input?.value?.checkValidity();
      return value;
    }

    function change(e) {
      context.emit('change', e.target.value);
      isInvalid.value = !input?.value?.checkValidity();
    }

    return {
      input,
      id,
      isInvalid,
      passwordView,
      iconClasses,
      glassBroken,
      classes,
      cssVars,
      defaultInputClasses,
      update,
      change,
    };
  },
});
</script>

<style scoped>
input:disabled {
  background-color: #f7f7f7;
}

textarea:disabled {
  background-color: #f7f7f7;
}

input::placeholder {
  @apply text-crisiscleanup-dark-200;
}

textarea::placeholder {
  @apply text-crisiscleanup-dark-200;
}

input.invalid {
  @apply border border-crisiscleanup-red-100;
}

textarea.invalid {
  @apply border border-crisiscleanup-red-100;
}

input {
  outline: none;
  width: var(--width);
  height: var(--height);
  border-radius: 0;
  @apply border border-crisiscleanup-dark-100;
  -webkit-appearance: none;
  opacity: 1;
}

textarea {
  outline: none;
  @apply border border-crisiscleanup-dark-100;
  border-radius: 0;
}

input:not([type='radio']):not([type='checkbox']) {
  -webkit-appearance: none;
  border-radius: 0;
}

input.xlarge {
  height: 60px;
}

input.large {
  height: 50px;
}

input.medium {
  height: 40px;
  width: 225px;
}

input.small {
  height: 32px;
  width: 150px;
}

textarea.large {
  height: 150px;
}

.icon-container {
  width: 40px;
  height: 40px;
  @apply border border-crisiscleanup-dark-100;
  border-left: 0;
}

.icon-container.large {
  width: 50px;
  height: 50px;
}

.icon-container.small {
  height: 30px;
}

.icon-container.has-tooltip {
  background-color: #f7f7f7;
}
</style>
