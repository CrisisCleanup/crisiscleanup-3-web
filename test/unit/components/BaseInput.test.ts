/**
 *
 * Tests for BaseInput
 *
 * Components
 */

import { describe, expect, it, vi } from 'vitest';
import { mount } from '@vue/test-utils';
import BaseInput from '@/components/BaseInput.vue';

describe('BaseInput', () => {
  it('should not log any errors', () => {
    const spy = vi.spyOn(global.console, 'error');
    mount(BaseInput);
    expect(spy).not.toHaveBeenCalled();
  });

  it('should emit value on input', () => {
    const wrapper = mount(BaseInput);
    const el = wrapper.find('input');
    el.setValue('new value');
    expect(wrapper.emitted().input).toMatchSnapshot();
  });

  it('should emit value on change', () => {
    const wrapper = mount(BaseInput);
    const el = wrapper.find('input');
    el.trigger('change');
    expect(wrapper.emitted().change.length).toBe(2);
  });

  it('should render correctly and match snapshot', () => {
    const testProps = {
      props: {
        size: 'large',
      },
    };
    const wrapper = mount(BaseInput, testProps);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render correctly and match snapshot with label', () => {
    const testProps = {
      props: {
        size: 'large',
        topLabel: 'Top Label',
      },
    };
    const wrapper = mount(BaseInput, testProps);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render correctly and match snapshot when required and provided', () => {
    const testProps = {
      props: {
        size: 'large',
        required: true,
        modelValue: 'Test',
      },
    };
    const wrapper = mount(BaseInput, testProps);
    expect(wrapper.element).toMatchSnapshot();
    wrapper.vm.update({
      target: {
        modelValue: 'Test',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.vm.isInvalid).toBe(false);
  });
});
