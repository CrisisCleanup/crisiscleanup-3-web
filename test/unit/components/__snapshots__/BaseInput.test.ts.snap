// Vitest Snapshot v1

exports[`BaseInput > should emit value on input 1`] = `undefined`;

exports[`BaseInput > should render correctly and match snapshot 1`] = `
<div
  class="flex justify-start relative items-center"
  data-v-82c69faf=""
  data-v-inspector-column="3"
  data-v-inspector-file="/Users/tobi/code/crisiscleanup-4-web/src/components/BaseInput.vue"
  data-v-inspector-line="2"
  data-v-inspector-title="BaseInput.vue"
>
  <!--v-if-->
  <input
    autocomplete="off"
    class="flex-grow p-1 text-base large"
    data-cy=""
    data-v-82c69faf=""
    pattern=".*"
    placeholder=""
    rows="0"
    style="--height: 40px; --width: 300px;"
    type="search"
  />
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
  
  
</div>
`;

exports[`BaseInput > should render correctly and match snapshot when required and provided 1`] = `
<div
  class="flex justify-start relative items-center"
  data-v-82c69faf=""
  data-v-inspector-column="3"
  data-v-inspector-file="/Users/tobi/code/crisiscleanup-4-web/src/components/BaseInput.vue"
  data-v-inspector-line="2"
  data-v-inspector-title="BaseInput.vue"
>
  <!--v-if-->
  <input
    autocomplete="off"
    class="flex-grow p-1 text-base large"
    data-cy=""
    data-v-82c69faf=""
    pattern=".*"
    placeholder=""
    required=""
    rows="0"
    style="--height: 40px; --width: 300px;"
    type="search"
  />
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
  
  
</div>
`;

exports[`BaseInput > should render correctly and match snapshot when required and provided 2`] = `
<div
  class="flex justify-start relative items-center"
  data-v-82c69faf=""
  data-v-inspector-column="3"
  data-v-inspector-file="/Users/tobi/code/crisiscleanup-4-web/src/components/BaseInput.vue"
  data-v-inspector-line="2"
  data-v-inspector-title="BaseInput.vue"
>
  <!--v-if-->
  <input
    autocomplete="off"
    class="flex-grow p-1 text-base large"
    data-cy=""
    data-v-82c69faf=""
    pattern=".*"
    placeholder=""
    required=""
    rows="0"
    style="--height: 40px; --width: 300px;"
    type="search"
  />
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
  
  
</div>
`;

exports[`BaseInput > should render correctly and match snapshot with label 1`] = `
<div
  class="flex justify-start relative flex-col items-start border"
  data-v-82c69faf=""
  data-v-inspector-column="3"
  data-v-inspector-file="/Users/tobi/code/crisiscleanup-4-web/src/components/BaseInput.vue"
  data-v-inspector-line="2"
  data-v-inspector-title="BaseInput.vue"
>
  <label
    class="text-xs px-1 text-crisiscleanup-dark-300"
    data-v-82c69faf=""
    data-v-inspector-column="5"
    data-v-inspector-file="/Users/tobi/code/crisiscleanup-4-web/src/components/BaseInput.vue"
    data-v-inspector-line="3"
    data-v-inspector-title="BaseInput.vue"
  >
    Top Label
  </label>
  <input
    autocomplete="off"
    class="flex-grow p-1 text-base large border-none"
    data-cy=""
    data-v-82c69faf=""
    pattern=".*"
    placeholder=""
    rows="0"
    style="--height: 40px; --width: 300px;"
    type="search"
  />
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
  
  
</div>
`;
